main/SQLHelper.cpp

```diff
@@ -3553,12 +3556,34 @@ uint64_t CSQLHelper::UpdateValueInt(const int HardwareID, const char* ID, const
 					(sValue == old_sValue)
 					);
 			}
+			else if (stype == STYPE_BlindsPercentageInverted)
+			{
+				int level = 0;
+				std::stringstream ssLevel(sValue);
+				ssLevel >> level;
+
+				if (nValue == gswitch_sOn)
+					level = 0;
+				else if (nValue == gswitch_sOff)
+					level = 100;
+				else
+				{
+					if (level == 100)
+						nValue2 = gswitch_sOff;
+					else
+						level = 100 - level; // received an inverted level, inverting back to original
+				}
+				ssLevel.str(std::string());
+				ssLevel.clear();
+				ssLevel << level;
+				ssLevel >> sValue2;
+			}
 
 			result = safe_query(
 				"UPDATE DeviceStatus SET SignalLevel=%d, BatteryLevel=%d, nValue=%d, sValue='%q', LastUpdate='%04d-%02d-%02d %02d:%02d:%02d' "
 				"WHERE (ID = %" PRIu64 ")",
 				signallevel, batterylevel,
-				nValue, sValue,
+				nValue2, sValue2.c_str(),
```

Comment:
This is quite unclear now... (at least... i find this unclear nvalue2/svalue2)
That is why i said it might be better to do this in the mainworker ?
No need to adjust any code in SQLHelper.cpp